### PR TITLE
DPL-827 scRNA Core Cell Extraction - added tube metadata for cell count, viability and volume

### LINKED ISSUE #1331: DPL-827 Save data against LRC Bank Stock tubes
**User story**
As a user of the scRNA Core Banking pipeline, I would like some key data to be saved against the LRC Bank Stock tubes, so that I can more easily perform the data transfer to the BioResource team in https://github.com/sanger/limber/issues/1439

**Who are the primary contacts for this story**
Lesley, Laura, Katy

**Who is the nominated tester for UAT**
Lesley, Laura (or, might be more of a technical story that we test within the team - it will then be tested when we retrieve the data in DPL-812)

**Acceptance criteria**
When the LRC PBMC Bank plate --> LRC Bank Stock tubes transfer takes place:
- [x] Cell count is stored against the LRC Bank Stock tubes as the sum of the cell counts from the source wells.
- [x] Viability is stored against the LRC Bank Stock tubes as the average (mean) of the viability from the source wells.
- [x] Volume is stored against the LRC Bank Stock tubes as the number of source wells multiplied by the volume (TBC) transferred by the robot each time 

**Dependencies**
This story is blocked by the following dependencies:
- https://github.com/sanger/limber/issues/1314

**Additional context**
Cell count and viability data is stored in the QC Results table.
Volume data is stored against the receptacle (well).

### PR DESCRIPTION
**Pull request by andrewsparkes**
Closes #1331 

#### Changes proposed in this pull request
Adds tube metadata

### CODE REVIEW

File: app/models/labware_creators/plate_split_to_tube_racks.rb [hunk 1]
```diff
@@ -540,45 +576,105 @@ def find_child_tube(well)
...
+      add_contingency_tube_metadata(tubes_metadata)
+      add_cell_count_metadata(tubes_metadata)
+      write_metadata_to_tubes(tubes_metadata)
     end
 
     # Adds tube rack barcode and position metadata to child sequencing tubes.
     #
     # @return [void]
-    def add_sequencing_tube_metadata
+    def add_sequencing_tube_metadata(tubes_metadata)
```
Comment (KatyTaylor) [inline app/models/labware_creators/plate_split_to_tube_racks.rb:594]: Code climate had a point 😆 there is some code duplication here. Maybe GitHub copilot can refactor it for you?

File: app/models/labware_creators/plate_split_to_tube_racks.rb [hunk 2]
```diff
@@ -540,45 +576,105 @@ def find_child_tube(well)
...
+
     # Shared method for adding tube rack barcode and position metadata to child tubes.
     #
     # @param child_tube [Tube] The child tube to add metadata to.
     # @param tube_posn [String] The position of the child tube in the tube rack.
     # @param tube_details [Hash] The tube details hash from the tube rack scan file.
     # @return [void]
-    def add_tube_metadata(child_tube, tube_posn, tube_details)
+    def update_metadata_on_tube(tube_barcode, metadata)
+      # TODO: why do I have to use to_s on volume here when already a string?
```
Comment (KatyTaylor) [inline app/models/labware_creators/plate_split_to_tube_racks.rb:668]: Ruby must be sneakily treating it as an integer?

**Review by KatyTaylor — APPROVED** [review]
Approved, but we should back out some of the code if needed after Stephen has done his story.

**Review by yoldas — APPROVED** [review]

**Comment by andrewsparkes** [comment]
Superceded by https://github.com/sanger/limber/issues/1439

**Pull request closed without merging**